lib/api/controllers/documentController.js

```diff
@@ -69,8 +69,8 @@ class DocumentController extends NativeController {
    */
   async search (request) {
     const { from, size, scrollTTL, searchBody } = request.getSearchParams();
-    const index = request.input.args.index;
-    const collection = request.input.args.collection;
+    const index = request.getIndex(false);
+    const collection = request.getCollection(false);
```

Comment:
not quite high with boolean in function that change the behavior... When this is the case, it's show that you need another function in reality.
Maybe do another function (getRequiredIndex) ?

Reply:
With a default struct with the field `required`, it would achieve the same goal in a clean way I think